pkg/kubelet/allocation/allocation_manager_test.go

```diff
@@ -1020,6 +1020,271 @@ func TestHandlePodResourcesResizeWithSwap(t *testing.T) {
+				if cpuRequest > 1 {
+					return lifecycle.PodAdmitResult{
+						Admit:   false,
+						Reason:  "OutOfcpu",
+						Message: fmt.Sprintf("not enough CPUs available for pod %s/%s, requested: %d, available:1", attrs.Pod.Namespace, attrs.Pod.Name, cpuRequest),
+					}
+				}
+				return lifecycle.PodAdmitResult{Admit: true}
+			},
+			// pod is still admitted as allocated resources are considered during admission.
+			expectAdmit:             false,
+			admissionFailureReason:  "OutOfcpu",
+			admissionFailureMessage: "not enough CPUs available for pod ns1/pod1, requested: 2, available:1",
+			// allocated Resources state must not be updated.
+			expectedAllocatedResourcesState: map[types.UID]v1.ResourceList{pod1UID: cpu1Mem1G},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.InPlacePodVerticalScaling, tc.ipprFeatureGate)
+			allocationManager := makeAllocationManager(t, &containertest.FakeRuntime{}, []*v1.Pod{})
+
+			podForAllocation := func(uid types.UID, resources v1.ResourceList) *v1.Pod {
+				return &v1.Pod{
+					ObjectMeta: metav1.ObjectMeta{UID: uid},
+					Spec: v1.PodSpec{
+						Containers: []v1.Container{{
+							Name:      containerName,
+							Resources: v1.ResourceRequirements{Requests: resources},
+						}},
+					},
+				}
+			}
+
+			for podUID, resources := range tc.initialAllocatedResourcesState {
+				err := allocationManager.SetAllocatedResources(podForAllocation(podUID, resources))
+				require.NoError(t, err)
+			}
+
+			if tc.admitFunc != nil {
+				handler := &testPodAdmitHandler{admitFunc: tc.admitFunc}
+				allocationManager.AddPodAdmitHandlers(lifecycle.PodAdmitHandlers{handler})
+			}
+
+			ok, reason, message := allocationManager.AddPod(tc.currentActivePods, tc.podToAdd)
+			require.Equal(t, tc.expectAdmit, ok)
+			require.Equal(t, tc.admissionFailureReason, reason)
+			require.Equal(t, tc.admissionFailureMessage, message)
+
+			for podUID, resources := range tc.expectedAllocatedResourcesState {
+				pod := podForAllocation(podUID, resources)
+				for _, container := range pod.Spec.Containers {
+					allocatedResources, found := allocationManager.GetContainerResourceAllocation(pod.UID, container.Name)
+					if pod.UID == tc.podToAdd.UID {
+						if tc.expectAdmit && !found {
+							t.Fatalf("resource allocation should exist for pod: %s", tc.podToAdd.Name)
+						}
+						if !tc.expectAdmit && found {
+							initialResources := tc.initialAllocatedResourcesState[pod.UID]
+							// allocated resources should not be modified when the pod is not admitted
+							assert.Equal(t, initialResources, allocatedResources.Requests, tc.name)
+						}
+					}
+					assert.Equal(t, container.Resources, allocatedResources, tc.name)
+				}
+			}
+
+			// Undo current modifications
+			allocationManager.RemovePod(tc.podToAdd.UID)
```

Comment:
nit: Looks like you are making a new allocation manager for each subtest, why is this needed? Is there some sort of state that persists between tests?